Phase diagram (#4252)
geodynamics/aspect: doc/manual/cookbooks/phase_diagram/doc/phase_diagram.tex

This is assured by assigning a constant density with zero expansivity.
It differs from a classic phase diagram only in the direction of pressure increase.

\par Two compositions of pyrolite and harzburgite are included in the model. % composition
With each run, only one composition is assigned to the whole domain in order to visualize the diagrams of these two separately.

\par The material model is then set up to mimic mantle phase transitions at 410, 520, and 660 km.% setup of mantle transitions
Details of phase transitions are taken from \cite{billen2018decoupling}.
A trick of assigning the values of density to the field of thermal capacity is used here [MIB: How is this done? can you be more explicit]. % trick
This serves the goal of visualizing values of reference densities of phases with the 'real density' assigned with a constant value in the model.
Inputs for this material model is listed here:
\lstinputlisting[language=prmfile]{cookbooks/phase_diagram/doc/material_model.prm.out}

\paragraph{Results}

\par Visualization of the model results yields a phase diagram of a pyrolitic mantle (Figure \ref{fig:phase_diagram_ph_density}). % phase diagram

> The field shown here has the reference densities of the pyrolite phases, though settings of phase transitions are over-simplified.

Reviewer: What do you mean by "settings"? Can you clarify?

One may notice that 3 transitions (i.e. one in the olivine system, two in the spinel system) are included for the 660 interfaces, % 3 transitons, higher T

> and they need to be modified at a higher temperature.

Reviewer: Can you explain this a bit more? I don't really understand what you mean in this sentence.

In spite of the complexities of mantle phases, the focus of this first example is to simply illustrate this approach of visualizing it.
Beyond showing the diagram, I have also used the 'lineout' feature in VISIT to export the data along two vertical lines at T = 1173 and T = 1673 (Figure \ref{fig:phase_diagram_ph_profile}). % linear profile
The T = 1173 illustrates the buoyancy forces felt by a descending cold slab within the mantle transition zone.

\par Next, we shift to harzburgite by changing the values of the initial compositional field from 0 to 1.  % harzburgite
For this to occur, following changes are needed:

> \lstinputlisting[language=prmfile]{cookbooks/phase_diagram/doc/harzburgite.prm.out}

Reviewer: Here you can also mention the other input file I am asking you to add.

With this change, we can also visualize the phase diagram of harzburgite in Figure \ref{fig:phase_diagram_ph_density}.

> 

Reviewer: I would add a new paragraph here, maybe something like \paragraph{The Steinberger case: Using a lookup table}

\par Moreover, I tested the a pyrolitic lookup table used in the Steinberg material model (Figure \ref{fig:phase_diagram_steinberg_density}).% steinberg model, todo fix reference

> The same setup of the initial condition is applied as in the previous case. % initial condition

Reviewer: ```suggestion
We use the same setup of the initial conditions as in the previous case. 
```

The densities, however, are not assigned to the heat capacity anymore.

> Thus the vertical axis would deviate from the axis of pressure a little bit.

Reviewer: ```suggestion
Thus the pressure gradient is not constant any more, and the vertical axis slightly deviates from the axis of pressure in the previous model.
```
Can you mention somewhere how different the maximum pressure is?

> This serves the goal of illustrating a more complex and thus more realistic model of phase transitions.

Reviewer: ```suggestion
This second setup serves the goal of illustrating a more complex and thus more realistic model of phase transitions.
```

Modification for the material model is listed below:
\lstinputlisting[language=prmfile]{cookbooks/phase_diagram/doc/steinberg.prm.out}  % todo add 

> 

Reviewer: I think it would be nice to have one or two sentences about how the phase diagrams compare, but I will leave this up to you.
